lib/kvdb/ikvdb.c

```diff
@@ -2494,7 +2496,7 @@ ikvdb_kvs_put(
     vbufsz = tls_vbufsz;
     vbuf = NULL;
 
-    if (clen == 0 && vlen > kk->kk_vcompmin && !(flags & HSE_KVS_PUT_VCOMP_OFF)) {
+    if (clen == 0 && vlen > CN_SMALL_VALUE_THRESHOLD && is_compression_allowed(kk, flags)) {
```

Comment:
I still think we should assert clen == 0 here and remove it from the conditional

Reply:
sure.  i can always put it back if i need to implement dump/restore of compressed values and come through this function.
i think this was only here because at one time wal needed to come through here with compressed values...

Reply:
Just gonna leave this in for now.